### PR TITLE
Compatible with most recent yosys

### PR DESCRIPTION
**Pull request by rowanG077**
Main change is that Yosys changed how ranges are stored.

### CODE REVIEW

File: frontends/systemverilog/uhdm_ast.cc [hunk 1]
```diff
@@ -1116,8 +1110,9 @@ static int range_width_local(AST::AstNode *node, AST::AstNode *rnode)
 
 static void save_struct_array_width_local(AST::AstNode *node, int width)
 {
+    // TODO: What to do with this?
     // stash the stride for the array
-    node->multirange_dimensions.push_back(width);
+    // node->multirange_dimensions.push_back(width);
```
Comment (rowanG077) [inline frontends/systemverilog/uhdm_ast.cc:1115]: I didn't know what to do with this. It seems wrong even to store the width here. But I couldn't follow what's going on so I have commented it out. If anyone know what's going on here please let me know.

**Comment by rowanG077** [comment]
This is a large change. I'm not sure how to approach keeping compatibility with older yosys versions. Factoring everything it would be a lot of work.

**Comment by alaindargelas** [comment]
Nice!
Maintainers need to approve the workflow

**Comment by rowanG077** [comment]
It still uses the old yosys in CI. Where can I update this?

**Comment by tgorochowik** [comment]
Hi, thanks for the PR!

Don't worry about backwards compatibility, as long as all the tests pass with newer yosys we don't have to care about older versions.

You can update yosys by changing the submodule here: https://github.com/chipsalliance/synlig/tree/main/third_party

**Comment by alaindargelas** [comment]
@rowanG077 , this PR from Yosys: https://github.com/YosysHQ/yosys/commit/1bbea13f80fa7e5d337dcf3affd28d1ab8664cae#diff-9cf03955fa6ed7051b83cfb95de9e3c42eeeecdff3fa071fbb14ac43e41b12a0R4686

seems to be a problem with the current state of Synlig, can you resubmit your Yosys update and point Yosys to the version right before that PR?

Your current Action run shows large amounts of failures. We already noticed separately that this particular PR is a problem when selectively added to Yosys 0.38 and used with Synlig

**Comment by rowanG077** [comment]
I'm sorry I don't have the bandwidth to keep pinging here and waiting everytime CI fails. This is an unworkable solution for me and there seem to be checks I cannot run locally.

**Comment by chili-chips-ba** [comment]
Is the plan for this merge to go ahead?
  Also see this discussion [thread](https://github.com/YosysHQ/yosys/issues/4436#issuecomment-2158002696)...

**Comment by tgorochowik** [comment]
The changes introduced in this PR look mostly alright, but as you can see the CI is failing, we cannot merge this until all of the tests are fixed and the CI is passing.

Feel free to take over if you'd like to see it merged sooner

**Comment by kamilrakoczy** [comment]
Thank you for working on this.
We managed to update yosys to version 0.44 (current upstream): https://github.com/chipsalliance/synlig/pull/2510
It included changes very similar to yours. 

This PR is no longer needed. Closing.

**Pull request closed without merging**